modules/aioseop_sitemap.php

```diff
@@ -403,7 +446,175 @@ public function __construct() {
+					}
+
+					$all_terms = array();
+
+					foreach ( $taxonomies as $taxonomy ) {
+						$terms = get_terms(
+							array(
+								'taxonomy' => $taxonomy,
+								'fields' => 'id=>name',
+								'hide_empty' => false,
+								'name__like' => $_POST['term'],
+								'order' => 'DESC',
+							)
+						);
+
+						if ( is_wp_error( $terms ) ) {
+							continue;
+						}
+
+						if ( $terms ) {
+							foreach ( $terms as $id => $term ) {
+								// show each term as a combination of term name and term taxonomy name.
+								$all_terms[] = array( 'id' => $id, 'name' => sprintf( '%s (%s)', $term, $taxonomy ) );
+							}
+						}
+					}
+					if ( empty( $all_terms ) ) {
+						wp_send_json_error( array( 'msg' => __( 'No terms found', 'all-in-one-seo-pack' ) ) );
+					}
+					$response = $all_terms;
+					break;
+			}
+			wp_send_json_success( $response );
+		}
+
+		/**
+		 * Generates the values to be displayed as pre-selected in the exclude terms box.
+		 *
+		 * @param array $terms Array of term_ids.
+		 *
+		 * @since 3.0
+		 *
+		 * @return array Array of term_id vs term_name/term_taxonomy combination.
+		 */
+		private function get_terms_as_selected( $terms ) {
+			$all_terms = array();
+			foreach ( $terms as $id ) {
+				$term = get_term( (int) $id );
+				if ( ! $term ) {
+					continue;
+				}
+				$all_terms[ $id ] = sprintf( '%s (%s)', $term->name, $term->taxonomy );
+			}
+			return $all_terms;
+		}
+
+
+		/**
+		 * Checks if a taxonomy has been excluded. This could only be because all its terms have been selected to be excluded.
+		 *
+		 * @since 3.0
+		 *
+		 * @param string    $tax    The taxonomy name.
+		 *
+		 * @return bool Whether a taxonomy has a been excluded.
+		 */
+		private function has_taxonomy_been_excluded( $tax ) {
+			global $wp_version;
+			$term_count = intval( wp_count_terms( $tax, array( 'hide_empty' => true ) ) );
+			if ( 0 === $term_count ) {
+				// no terms in this taxonomy, so not excluded.
+				return false;
+			}
+
+			// terms excluded in the settings.
+			$terms_excluded = $this->options[ "{$this->prefix}excl_terms" ];
+			if ( empty( $terms_excluded ) ) {
+				// no terms excluded from any category, so not excluded.
+				return false;
+			}
+
+			// terms defined for this taxonomy.
+			$terms  = get_terms(
+				array(
+					'taxonomy' => $tax,
+					'hide_empty' => true,
+				)
+			);
+			if ( is_wp_error( $terms ) ) {
+				return false;
+			}
+			$term_ids = array();
+			foreach ( $terms as $term ) {
+				$term_ids[] = $term->term_id;
+			}
+			// are all terms from this taxonomy excluded?
+			$common = count( array_intersect( $term_ids, $terms_excluded ) );
+			// the category is excluded if ALL of its terms have been specified in the exclude terms list.
+			return $common === count( $term_ids );
+		}
+
+		/**
+		 * Load resources.
+		 *
+		 * @since 3.0
+		 *
+		 * @param string $hook_suffix The current admin page.
+		 */
+		function admin_enqueue_scripts( $hook_suffix ) {
```

Comment:
This would either need to be renamed, or call the parent method; this would break the Design Pattern when working in conjunction with the parent class.

Reply:
have renamed the function

Reply:
I just tested this in conjunction with the parent method, and it works as intended. It would be better to use that to keep everything tied together rather than invent an alternate method.

I've uploaded a commit showing the changes https://github.com/EkoJR/all-in-one-seo-pack/commit/998cd74b2a6ba25dfdf45b63f4401c8ec61d8019

Reply:
I think its alright this way without violating any conventions. Interestingly, this has been used in many other classes. I understand you would have done it that way instead of this way but that's fair.

Reply:
There's no reason to create an alternate function like this. I tested and it does work with the parent as intended, and also follows the encapsulation pattern.